woocommerce/woocommerce
# Fix incorrect VAT exempt behaviour on shop page when prices are exclusive of tax.

### All Submissions:


**Note: This PR superceed the other PR #27020 by adding unit tests. ( #27020 is older and require rebase. So, I included changes of that PR in this PR as well)**

### Changes proposed in this Pull Request:
As described on #27012, VAT from product prices on shop pages is not getting removed for vat exempted customers, if prices are entered exclusive of tax. It removes VAT only if the shop setting is "I will enter prices inclusive of tax". 

Closes #27012

### How to test the changes in this Pull Request:

Please see more detailed instructions in the [comment below](https://github.com/woocommerce/woocommerce/pull/33991#issuecomment-1451924872).

1. Set VAT exempt on the current user using one of the below ways.
     A. Install [EU VAT Number](https://woocommerce.com/products/eu-vat-number/) and save the correct VAT number from My Account. 
           - OR -
     B. Set vat exempt programmatically:  
     ```
     add_action('init', function(){
       if (isset(WC()->customer)){
          WC()->customer->set_is_vat_exempt(true);
       }
    });
     ```
2. Set tax options like this in Woo settings
![image](https://user-images.githubusercontent.com/10613171/179961522-2cc7fafd-a571-4ac2-a816-bdfcd459f236.png)
3. Create a product with a price of 100 and a tax rate of 15% for testing purposes.
4. Visit the Shop page and make sure the price for the created product showing 100.

### Other information:




### FOR PR REVIEWER ONLY:
